src/functions.c

```diff
@@ -370,6 +370,61 @@ void fcallCommandReadOnly(client *c) {
     fcallCommandGeneric(c, 1);
 }
 
+void functionsDumpCommand(client *c) {
+    unsigned char buf[2];
+    uint64_t crc;
+    rio payload;
+    rioInitWithBuffer(&payload,sdsempty());
+
+    functionsSaveRio(&payload);
+
+    /* RDB version */
+    buf[0] = RDB_VERSION & 0xff;
+    buf[1] = (RDB_VERSION >> 8) & 0xff;
+    payload.io.buffer.ptr = sdscatlen(payload.io.buffer.ptr,buf,2);
+
+    /* CRC64 */
+    crc = crc64(0,(unsigned char*)payload.io.buffer.ptr,
+                sdslen(payload.io.buffer.ptr));
+    memrev64ifbe(&crc);
+    payload.io.buffer.ptr = sdscatlen(payload.io.buffer.ptr,&crc,8);
+
+    addReplyBulkSds(c,payload.io.buffer.ptr);
+}
+
+void functionsRestoreCommand(client *c) {
+    uint16_t rdbver;
+    if (verifyDumpPayload(c->argv[2]->ptr, sdslen(c->argv[2]->ptr), &rdbver) != C_OK) {
+        addReplyError(c,"DUMP payload version or checksum are wrong");
```

Comment:
We are just checking the version here right?

Reply:
it should also verify the checksum.

Reply:
version + checksum